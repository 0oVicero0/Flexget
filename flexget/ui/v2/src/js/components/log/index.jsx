import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Header from 'containers/log/Header';
import LogTable from 'containers/log/LogTable';

const styleSheet = theme => ({
  root: {
    padding: 24,
    display: 'flex',
    height: '100%',
    boxSizing: 'border-box',
    flexDirection: 'column',
    [theme.breakpoints.up('sm')]: {
      paddingTop: 0,
    },
  },
  logTable: {
    width: 'initial',
    flex: {
      grow: 1,
    },
  },
});

const LogPage = ({ classes }) => (
  <Paper className={classes.root} elevation={4}>
    <Header />
    <div className={classes.logTable}>
      <LogTable />
    </div>
  </Paper>
);

LogPage.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styleSheet)(LogPage);
