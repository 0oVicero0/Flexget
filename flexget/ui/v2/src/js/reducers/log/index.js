import {
  LOG_START,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_LINES,
  LOG_QUERY,
  LOG_CLEAR,
} from 'actions/log';

const initState = {
  messages: [],
  connected: false,
  lines: '200',
  query: '',
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case LOG_START:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case LOG_LINES:
      return {
        ...state,
        lines: action.payload,
      };
    case LOG_QUERY:
      return {
        ...state,
        query: action.payload,
      };
    case LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
