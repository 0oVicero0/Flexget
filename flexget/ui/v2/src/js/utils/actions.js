export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
