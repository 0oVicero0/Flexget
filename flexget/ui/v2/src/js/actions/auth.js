import * as utils from 'utils/actions';
import { get, post } from 'utils/fetch';

export const AUTH = 'AUTH';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

const createAction = utils.createAction(AUTH);
const loading = utils.createLoading(AUTH);

export function checkLogin() {
  return (dispatch) => {
    get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
  };
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    get('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .then(err => dispatch(createAction(LOGOUT, err)));
  };
}
