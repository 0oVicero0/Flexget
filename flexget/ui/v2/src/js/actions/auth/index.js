import { createAction, loading } from 'utils/actions';
import { get, post } from 'utils/fetch';

const PREFIX = '@flexget/auth/';
export const LOGIN = `${PREFIX}LOGIN`;
export const LOGOUT = `${PREFIX}LOGOUT`;

export function checkLogin() {
  return dispatch => get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    return post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    return post('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .catch(err => dispatch(createAction(LOGOUT, err)));
  };
}
