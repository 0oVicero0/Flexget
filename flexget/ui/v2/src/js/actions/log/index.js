import oboe from 'oboe';
import { createAction } from 'utils/actions';

const PREFIX = '@flexget/log/';
export const LOG_START = `${PREFIX}LOG_START`;
export const LOG_MESSAGE = `${PREFIX}LOG_MESSAGE`;
export const LOG_DISCONNECT = `${PREFIX}LOG_DISCONNECT`;
export const LOG_LINES = `${PREFIX}LOG_LINES`;
export const LOG_QUERY = `${PREFIX}LOG_QUERY`;
export const LOG_CLEAR = `${PREFIX}LOG_CLEAR`;

export function startLogStream() {
  return (dispatch, getState) => {
    const { lines, query } = getState().log;
    const stream = oboe({
      url: `api/server/log?lines=${lines}&search=${query}`,
      method: 'GET',
    });

    const promise = new Promise((resolve, reject) => {
      stream
        .on('start', () => dispatch(createAction(LOG_START)))
        .on('node', '{message}', message => dispatch(createAction(LOG_MESSAGE, message)))
        .done(() => resolve())
        .fail(err => reject(err));
    });

    promise.abort = () => {
      stream.abort();
      dispatch(createAction(LOG_DISCONNECT));
    };

    return promise;
  };
}

export function setLines(lines) {
  return createAction(LOG_LINES, lines);
}

export function setQuery(query) {
  return createAction(LOG_QUERY, query);
}

export function clearLogs() {
  return createAction(LOG_CLEAR);
}

