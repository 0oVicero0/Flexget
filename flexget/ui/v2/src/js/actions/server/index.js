import { createAction, loading } from 'utils/actions';
import { post } from 'utils/fetch';

const PREFIX = '@flexget/server/';
export const SERVER_RELOAD = `${PREFIX}SERVER_RELOAD`;
export const SERVER_SHUTDOWN = `${PREFIX}SERVER_SHUTDOWN`;


function manageServer(action, operation) {
  return (dispatch) => {
    dispatch(loading(action));
    return post('/server/manage', { operation })
      .then(({ data }) => dispatch(createAction(action, {}, { message: data.message })))
      .catch(err => dispatch(createAction(action, err)));
  };
}

export function reloadServer() {
  return manageServer(SERVER_RELOAD, 'reload');
}

export function shutdownServer() {
  return manageServer(SERVER_SHUTDOWN, 'shutdown');
}
