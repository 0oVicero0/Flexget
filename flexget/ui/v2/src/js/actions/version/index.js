import * as utils from 'utils/actions';
import { get } from 'utils/fetch';

export const VERSION = 'VERSION';
export const GET_VERSION = 'GET_VERSION';

const createAction = utils.createAction(VERSION);
const loading = utils.createLoading(VERSION);

export function getVersion() {
  return (dispatch) => {
    dispatch(loading(GET_VERSION));
    return get('/server/version')
      .then(version => dispatch(createAction(GET_VERSION, version)))
      .catch(err => dispatch(createAction(GET_VERSION, err)));
  };
}
