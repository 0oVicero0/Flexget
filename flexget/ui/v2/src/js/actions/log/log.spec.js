import configureMockStore from 'redux-mock-store';
import * as log from 'actions/log';

const mockStore = configureMockStore();

describe('actions/log', () => {
  describe('setLines', () => {
    it('dispatches the correct acction when setting lines', () => {
      const store = mockStore({ log: {} });

      store.dispatch(log.setLines(400));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('setQuery', () => {
    it('dispatches the correct acction when setting query', () => {
      const store = mockStore({ log: {} });

      store.dispatch(log.setQuery(400));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('clearLogs', () => {
    it('dispatches the correct acction when clearing logs', () => {
      const store = mockStore({ log: {} });

      store.dispatch(log.clearLogs(400));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
