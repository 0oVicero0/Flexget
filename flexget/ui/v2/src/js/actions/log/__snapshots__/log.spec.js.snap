// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/log clearLogs dispatches the correct acction when clearing logs 1`] = `
Array [
  Object {
    "error": false,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "LOG",
    },
    "payload": undefined,
    "type": "LOG_CLEAR",
  },
]
`;

exports[`actions/log setLines dispatches the correct acction when setting lines 1`] = `
Array [
  Object {
    "error": false,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "LOG",
    },
    "payload": 400,
    "type": "LOG_LINES",
  },
]
`;

exports[`actions/log setQuery dispatches the correct acction when setting query 1`] = `
Array [
  Object {
    "error": false,
    "meta": Object {
      "ignore": false,
      "message": undefined,
      "namespace": "LOG",
    },
    "payload": 400,
    "type": "LOG_QUERY",
  },
]
`;
